### PR TITLE
add libuv and cmocka test cases with file system on sim and qemu on CI

### PR DESCRIPTION
**Pull request by vela-mib**
## Summary
add libuv and cmocka test cases with file system on sim and qemu on CI
## Impact
CI run
## Testing
localhost

### CODE REVIEW

**Comment by acassis** [comment]
Please update this page: https://nuttx.apache.org/docs/latest/platforms/risc-v/qemu-rv/boards/rv-virt/index.html to include these new tests. Please include the steps to run these python scripts, the qemu parameters, etc

**Comment by vela-mib** [comment]
> Please update this page: https://nuttx.apache.org/docs/latest/platforms/risc-v/qemu-rv/boards/rv-virt/index.html to include these new tests. Please include the steps to run these python scripts, the qemu parameters, etc

done.

**Review by acassis — APPROVED** [review]

**Comment by acassis** [comment]
Please fix these errors:
```
kernel/fs/cases/fs_eventfd_test.c: In function 'thread_func':
Error: kernel/fs/cases/fs_eventfd_test.c:50:12: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
   50 |       read((int)args, &eventfd01_buffer, sizeof(eventfd_t));
      |            ^
cc1: all warnings being treated as errors
make[2]: *** [/github/workspace/sources/apps/Application.mk:200: kernel/fs/cases/fs_eventfd_test.c.github.workspace.sources.apps.testing.testsuites.o] Error 1
kernel/fs/cases/fs_fsync_test.c: In function 'test_nuttx_fs_fsync02':
Error: kernel/fs/cases/fs_fsync_test.c:120:34: error: format '%d' expects argument of type 'int', but argument 3 has type 'size_t' {aka 'long unsigned int'} [-Werror=format=]
  120 |   syslog(LOG_INFO, "the bsize = %d\n", statfsbuf.f_bsize);
      |                                 ~^     ~~~~~~~~~~~~~~~~~
      |                                  |              |
      |                                  int            size_t {aka long unsigned int}
      |                                 %ld
cc1: all warnings being treated as errors
make[2]: *** [/github/workspace/sources/apps/Application.mk:200: kernel/fs/cases/fs_fsync_test.c.github.workspace.sources.apps.testing.testsuites.o] Error 1
kernel/fs/cases/fs_stream_test.c: In function 'test_nuttx_fs_stream03':
Error: kernel/fs/cases/fs_stream_test.c:270:61: error: format '%zi' expects argument of type 'signed size_t', but argument 4 has type 'int' [-Werror=format=]
  270 |                  "len = %zi != return value from fwrite = %zi",
      |                                                           ~~^
      |                                                             |
      |                                                             long int
      |                                                           %i
  271 |                  len, ret);
      |                       ~~~                                    
      |                       |
      |                       int
Error: kernel/fs/cases/fs_stream_test.c:303:60: error: format '%zi' expects argument of type 'signed size_t', but argument 4 has type 'int' [-Werror=format=]
  303 |                  "len = %zi != return value from fread = %zi",
      |                                                          ~~^
      |                                                            |
      |                                                            long int
      |                                                          %i
  304 |                  len, ret);
      |                       ~~~                                   
      |                       |
      |                       int
cc1: all warnings being treated as errors
make[2]: *** [/github/workspace/sources/apps/Application.mk:200: kernel/fs/cases/fs_stream_test.c.github.workspace.sources.apps.testing.testsuites.o] Error 1
kernel/time/cases/clock_test_timer03.c: In function 'test_nuttx_clock_test_timer03':
Error: kernel/time/cases/clock_test_timer03.c:126:39: error: format '%lld' expects argument of type 'long long int', but argument 3 has type 'int64_t' {aka 'long int'} [-Werror=format=]
  126 |       syslog(LOG_INFO, "expected = %lld escaped = %lld failed = %d",
      |                                    ~~~^
      |                                       |
      |                                       long long int
      |                                    %ld
  127 |              expected, escaped, failed);
      |              ~~~~~~~~                  
      |              |
      |              int64_t {aka long int}
Error: kernel/time/cases/clock_test_timer03.c:126:54: error: format '%lld' expects argument of type 'long long int', but argument 4 has type 'int64_t' {aka 'long int'} [-Werror=format=]
  126 |       syslog(LOG_INFO, "expected = %lld escaped = %lld failed = %d",
      |                                                   ~~~^
      |                                                      |
      |                                                      long long int
      |                                                   %ld
  127 |              expected, escaped, failed);
      |                        ~~~~~~~                        
      |                        |
      |                        int64_t {aka long int}
cc1: all warnings being treated as errors
```

**Review by xiaoxiang781216 — APPROVED** [review]

File: tools/ci/cirun.sh [hunk 1]
```diff
@@ -42,6 +42,7 @@ find ${nuttx}/tools/ci/testrun -name '.pytest_cache' |xargs rm -rf
 rm -rf ${logs}
 rm -rf ${nuttx}/../apps/testing/ltp/ltp/
 rm -f ${nuttx}/../apps/testing/cmocka/cmocka.zip
```
Comment (xiaoxiang781216) [inline tools/ci/cirun.sh:44]: let's move cmocka.zip and ltp clearup to the related project under apps

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**